include/swift/AST/Types.h

```diff
@@ -2609,9 +2610,9 @@ class AnyFunctionType : public TypeBase {
   AnyFunctionType(TypeKind Kind, const ASTContext *CanTypeContext,
                   Type Input, Type Output, RecursiveTypeProperties properties,
                   unsigned NumParams, const ExtInfo &Info)
-  : TypeBase(Kind, CanTypeContext, properties), Input(Input), Output(Output),
-    NumParams(NumParams) {
+  : TypeBase(Kind, CanTypeContext, properties), Input(Input), Output(Output) {
     AnyFunctionTypeBits.ExtInfo = Info.Bits;
+    AnyFunctionTypeBits.NumParams = NumParams;
```

Comment:
Can you add an assertion that this hasn't dropped information? (in case someone actually has 1024 parameters)

Reply:
Sure. I figured that about 1000 parameters was about 100 times more than any reasonable programmer would need/want. :-)

Reply:
Oh, for sure, but code generators aren't reasonable programmers. (Clearly I don't *really* care at this time, though, or I'd ask for something stronger than an assertion.)

Reply:
Ya, code generators are a pain. They can easily exceed otherwise reasonable compile time limits. That being said, I think adding runtime errors when compile time limits are exceeded is a separable goal.